rasa/shared/core/domain.py

```diff
@@ -125,24 +125,67 @@ def empty(cls) -> "Domain":
         return cls([], [], [], {}, [], {})
 
     @classmethod
-    def load(cls, paths: Union[List[Union[Path, Text]], Text, Path]) -> "Domain":
-        if not paths:
+    def load(cls, path: Union[Text, Path]) -> "Domain":
+        """Loads the domain based on the dir/file path."""
+        if not path:
             raise InvalidDomain(
                 "No domain file was specified. Please specify a path "
                 "to a valid domain file."
             )
-        elif not isinstance(paths, list) and not isinstance(paths, set):
-            paths = [paths]
-
         domain = Domain.empty()
-        for path in paths:
-            other = cls.from_path(path)
-            domain = domain.merge(other)
+
+        # validate the domain file/s and return a domain_dict
+        # create a configured domain object from the domain_dict
+        # merge this into the empty domain
+        configured_domain_dict = cls.validate_and_merge_domain_files(path)
+        domain = domain.merge(cls.from_dict(configured_domain_dict))
 
         return domain
 
+    @staticmethod
+    def validate_and_merge_domain_files(path: Union[Text, Path]):
+        """Validates and merges the domain file/files based on the dir/file path."""
+        paths = []
+        if os.path.isfile(path):
+            paths.append(path)
+
+        if os.path.isdir(path):
+            for filename in os.listdir(path):
+                if filename.endswith(".yml"):
+                    path_to_file = os.path.join(path, filename)
+                    paths.append(path_to_file)
+
+        domains = []
+
+        for path in paths:
+            yaml = rasa.shared.utils.io.read_file(path)
+            try:
+                rasa.shared.utils.validation.validate_yaml_schema(
+                    yaml, DOMAIN_SCHEMA_FILE
+                )
+                data = rasa.shared.utils.io.read_yaml(yaml)
+                rasa.shared.utils.validation.validate_training_data_format_version(
+                    data, path
+                )
+                domains.append(data)
+            except (YamlException, TypeError) as e:
+                e.filename = path
+                raise e
+
+        domain_dict = collections.defaultdict(list)
```

Comment:
I think it makes sense to split this part further and add more simpler tests to see how multiple dictionaries are merged with respect to entities spread around different files, expected behaviour around `use_entities` or `ignore_entities` etc.